homeassistant/components/neato/config_flow.py

```diff
@@ -47,7 +54,17 @@ async def async_step_reauth_confirm(self, user_input: dict | None = None) -> dic
     async def async_oauth_create_entry(self, data: dict) -> dict:
         """Create an entry for the flow. Update an entry if one already exist."""
         current_entries = self._async_current_entries()
-        if current_entries and CONF_TOKEN not in current_entries[0].data:
+        if current_entries:
+            neato_session = self.hass.data[NEATO_DOMAIN][current_entries[0].entry_id]
+            hub = NeatoHub(self.hass, Account(neato_session))
+            data["unique_id"] = await hub.async_update_entry_unique_id(
```

Comment:
I suggest only updating the unique id during config entry setup, which we already seem to be doing.

Reply:
Removed